sympy/solvers/solveset.py

```diff
@@ -640,6 +640,93 @@ def _solveset(f, symbol, domain, _check=False):
     return result
 
 
+def check_noninverts(f, symbol):
+    """substitute the non invertable terms Derivative, f(x) etc.
```

Comment:
Please be specific and specify what "etc" means, you can simply say Derivative, Function and Integral.